zerver/lib/bugdown/__init__.py

```diff
@@ -1418,6 +1452,9 @@ def bugdown_stats_finish():
 
 def convert(content, message=None, message_realm=None, possible_words=None, sent_by_bot=False):
     # type: (Text, Optional[Message], Optional[Realm], Optional[Set[Text]], Optional[bool]) -> Text
+    global current_message_realm_id
+    if message_realm is not None:
+        current_message_realm_id = message_realm.id
```

Comment:
You don't need all this complexity; just access `message_realm`  I think you should factor the check into a function in `zerver/models.py` like this:

```
def image_preview_enabled_for_realm(realm):
    if not settings.INLINE_URL_EMBED_PREVIEW:
         return False
    if realm is None:
         return True
    return realm.inline_image_preview
```

and then these calls can just be `if image_preview_enabled_for_realm(message_realm)` or whatever